src/services/app-storage.ts

```diff
@@ -192,6 +192,63 @@ export class AppStorage {
 		});
 	}
 
+	/**
+	 * Delete the data for a player.
+	 *
+	 * @param {Player} player The player to delete.
+	 * @memberof AppStorage
+	 */
+	public deletePlayer(player: Player): void {
+
+		// Finds groups that deleted player is in and deletes the group
+		this.getGroups().then((groups: Group[]) => {
+
+			// Go through each group and see if the deleted player is in the group.
+			// If deleted player is in group, then delete group.
+			groups.forEach((group: Group) => {
+				group.playerIds.forEach((memberId) => {
+					if(player.storageId === memberId) {
+
+						// Delete this group because the player is being deleted
+						this.deleteGroup(group);
+					}
+				});
+			});
+
+			this.deleteObject<Player>(player, this.playersCollection);
+
```

Comment:
So many empty newlines. Guidelines for newlines: following closing blocks, following exit keywords (return, break), prior to comment lines. And MAYBE between blocks of code, but this really should be rare, most contiguous code either is meant to be contiguous or should be split into functions.